src/nrnpython/nrnpy_nrn.cpp

```diff
@@ -2207,71 +2200,73 @@ static PyObject* var_of_mech_next_safe(NPyVarOfMechIter* self) {
 
 static PyObject* segment_getattro(NPySegObj* self, PyObject* pyname) {
     Section* sec = self->pysec_->sec_;
-    CHECK_SEC_INVALID(sec)
+    CHECK_SEC_INVALID(sec);
 
-    Symbol* sym;
-    Py_INCREF(pyname);
+    auto _pyname_tracker = nb::borrow(pyname);  // keep refcount+1 during use
```

Comment:
Same as above. I'd like to split the ref-counting related changes from the rest.